src/XliffTasks.Tests/XamlRuleDocumentTests.cs

```diff
@@ -160,33 +678,205 @@ public void BasicLoadAndTranslate()
+      <NameValuePair Name=""TypeDescriptorText"" Value=""NNN"" xliff:LocalizedProperties=""Value"">
+        <!-- Value: My type descriptor text comment -->
+      </NameValuePair>
     </StringProperty.Metadata>
   </StringProperty>
 </Rule>";
 
-            XamlRuleDocument document = new();
-            StringWriter writer = new();
-            document.Load(new StringReader(source));
+            RunXamlTranslationTest(source, translations, expectedTranslation, expectedXlf);
+        }
+
+        
+        /* the purpose of this test is to ensure that deeply nested translations translate as expected, as well as attributes declared as a nested element:
+          such as:
+          <A Name="MyName">
+            <A.Description>MyName Description</A.Description>
+            <B>
+              <C>
+                <D>
+                  <NameValuePair Name="SearchTerms" TranslatableProp1="this-is-translated" xliff:LocalizedProperties="TranslatableProp1">
+                    <NameValuePair.Value>My Search Terms</NameValuePair.Value> 
+                </D>
+                ....
+    
+        */
+        [Fact]
+        public void LoadAndTranslateWithDeeplyNestedTranslationAndElementAttributeSyntax()
+        {
+            string source =
+@"<Rule Name=""MyRule""
+        DisplayName=""My rule display name""
+        PageTemplate=""generic""
+        Description=""My rule description""
+        xmlns=""http://schemas.microsoft.com/build/2009/properties"" xmlns:xliff=""https://github.com/dotnet/xliff-tasks"" xmlns:mc=""http://schemas.openxmlformats.org/markup-compatibility/2006""
+        mc:Ignorable=""xliff"">
+  <!-- DisplayName: My rule display name comment -->
+  <!-- Description: My rule description comment -->
+  <Rule.Categories>
+    <Category Name=""MyCategory"" Description=""My category description"">
+      <Category.DisplayName>My category display name</Category.DisplayName>
+      <!-- DisplayName: My category display name comment -->
+    </Category>
+  </Rule.Categories>
+  <StringProperty Name=""MyStringProperty"" DisplayName=""MyStringProperty display name"">
+    <StringProperty.Description>MyStringProperty description</StringProperty.Description>
+    <StringProperty.ValueEditors>
+      <ValueEditor EditorType=""MultiStringSelector"">
+        <ValueEditor.Metadata>
+          <NameValuePair Name=""TypeDescriptorText"" Value=""Custom symbols"" xliff:LocalizedProperties=""Value"" />
+          <NameValuePair Name=""SearchTerms"">
+            <NameValuePair.Value>My search terms</NameValuePair.Value>
+          </NameValuePair>
+
+          <TestElement Name=""TranslatableElement"" LocalizablePropC=""LocalizableC attribute"" xliff:LocalizedProperties=""LocalizablePropA;LocalizablePropB;LocalizablePropC"">
+            <TestElement.LocalizablePropA>LocalizableA descendent prop</TestElement.LocalizablePropA>
+            <TestElement.LocalizablePropB>LocalizableB descendent prop</TestElement.LocalizablePropB>
```

Comment:
Technically, property element syntax can be declared on a base type. For example, if class `C` derives from `Base`, and `Base` defines property `Property`:

```xml
<C>
  <Base.Property>Hello</Base.Property>
<C>
```

is equivalent to:

```xml
<C>
  <C.Property>Hello</C.Property>
<C>
```

We don't understand the type hierarchy of values defined here, so we can't easily obtain total parity. Options are to 1) leave it as is, 2) allow `*.Property` without requiring stuff before the `.` to match the parent.